### PR TITLE
Add Overwrite function

### PR DESCRIPTION
**Pull request by elb3k**
Add the `Override` function which will override the values in destination message using source message fields (filtered by paths)

### CODE REVIEW

File: fmutils.go [hunk 1]
```diff
@@ -150,3 +155,31 @@ func (mask NestedMask) Prune(msg proto.Message) {
 		return true
 	})
 }
+
+// Override - overrides all of the fields listed in paths in dest msg using the values from the src msg.
```
Comment (elb3k) [inline fmutils.go]: nit: better comment message

**Comment by mennanov** [comment]
It should be possible to use the `proto.Merge()` function https://github.com/protocolbuffers/protobuf-go/blob/8088bf85b8fc5a6c0d7e2a39743e237cae8bef99/proto/merge.go#L25 to achieve a similar result. Is there a reason you decided to add this function?

**Comment by elb3k** [comment]
> It should be possible to use the `proto.Merge()` function https://github.com/protocolbuffers/protobuf-go/blob/8088bf85b8fc5a6c0d7e2a39743e237cae8bef99/proto/merge.go#L25 to achieve a similar result. Is there a reason you decided to add this function?

Yes `proto.Merge()` can be used to achieve a similar result. But it only overrides filled fields (no 0, empty string nor nil message/list/map)
But this function only overrides the values set in the update mask and it supports empty values as well.

**Comment by elb3k** [comment]
My primary use case for this function is to override proto with a PATCH request.
`proto.Merge` works 95% of the time, but it does not distinguish between not set and set with zero value (0 for int, empty string for string field). Example:

Source message:
```golang
src := &testproto.Message{
  IntField: 32,
  StringField: "string",
  ListField: []string{"a", "b", "c"},
}
```

PATCH request:
```golang
patch := &testproto.Message{
  IntField: 0,
  StringField: "",
  ListField: []string{},
}

patchPaths := []string{"int_field", "string_field", "list_field"}
```

`proto.Merge` will not override those values, but the `Override` function will override every field that is in the `patchPaths`

File: fmutils_test.go [hunk 2]
```diff
@@ -768,6 +768,128 @@ func TestPrune(t *testing.T) {
...
+func TestOverride(t *testing.T) {
+	tests := []struct {
+		name  string
+		paths []string
+		src   proto.Message
+		dest  proto.Message
+		want  proto.Message
+	}{
+		{
+			name: "profile override",
```
Comment (mennanov) [inline fmutils_test.go]: Please add a better explanation for this test case. It's unclear what result is expected

File: fmutils.go [hunk 3]
```diff
@@ -21,6 +21,11 @@ func Prune(msg proto.Message, paths []string) {
 	NestedMaskFromPaths(paths).Prune(msg)
 }
 
+// Override - overrides all of the fields listed in paths in dest msg using the values from src msg.
+func Override(src, dest proto.Message, paths []string) {
```
Comment (mennanov) [inline fmutils.go]: How about renaming this function to `Overwrite`? What do you think?

File: fmutils.go (hunk 1 above)
Comment (mennanov) [reply fmutils.go]: Is this a TODO? If so, please address it.

**Review by mennanov — CHANGES_REQUESTED** [review]
I left a few comments to address, otherwise looks good!
Thanks!

File: fmutils.go (hunk 3 above)
Comment (elb3k) [reply fmutils.go]: yep, sg to me.

File: fmutils_test.go (hunk 2 above)
Comment (elb3k) [reply fmutils_test.go]: This test was a combination of a bunch of things, will separate it into few tests with better names

File: fmutils.go (hunk 1 above)
Comment (elb3k) [reply fmutils.go]: Addressed

**Review by mennanov — APPROVED** [review]

**Comment by mennanov** [comment]
Thanks a lot!

**Pull request merged**